src/plugins/interactive_setup/server/verification_code.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import chalk from 'chalk';
+import crypto from 'crypto';
+
+import type { Logger } from 'src/core/server';
+
+export class VerificationCode {
+  public readonly code = VerificationCode.generate();
+  private failedAttempts = 0;
+  private readonly maxFailedAttempts = 10;
```

Comment:
Agree that 10 seems a bit too much.
I'm not aware of any formula to arrive at a "correct" number of attempts, but a good rule of thumb is 3 attempts is enough. Especially since all they have to do if they get locked out is restart Kibana.

Reply:
Yeh, I wanted to avoid frustration since the required restart is quite annoying but happy to reduce that if it makes it fairly likely that the code can be guessed.

Reply:
As an example, iOS has an option to erase the phone after 10 failed passcode attempts. 

Possible permutations for 6 digit code is: 1,000,000

Reply:
ok we've settle for 5 attempts